rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -4,47 +4,53 @@
  */
 
 //! Protocol helpers.
-use crate::rejection::{ContentTypeRejection, MimeParsingFailed, MissingJsonContentType, MissingXmlContentType};
+use crate::rejection::RequestRejection;
 use axum_core::extract::RequestParts;
 
+#[derive(Debug)]
+pub enum Protocol {
+    RestJson1,
+    RestXml,
+}
```

Comment:
This should probably be non_exhaustive... I am also wondering if we need an "Unknown" variant that will signal us we got a protocol we do not support yet.. Thoughts?

Reply:
Ok, makes sense.